providers/aws/resources/aws.lr

```diff
@@ -121,6 +121,312 @@ private aws.vpc.flowlog @defaults("id region status") {
   tags map[string]string
 }
 
+// Amazon WAF v2
+aws.waf {
+  // List of WAF ACLs
+  acls() []aws.waf.acl
+  // List of WAF rules
+  ruleGroups() []aws.waf.rulegroup
+  // List of WAF IP sets
+  ipSets() []aws.waf.ipset
+}
+
+// Amazon WAF v2 ACL
+private aws.waf.acl @defaults("name") {
+  // ARN of the ACL
+  arn string
+  // ID of the ACL
+  id string
+  // Name of the ACL
+  name string
+  // Description of the ACL
+  description string
+  // Whether the ACL is managed by Firewall Manager
+  managedByFirewallManager bool
+  // List of WAF rules
+  rules() []aws.waf.rule
+}
+
+// Amazon WAF v2 RuleGroup
+private aws.waf.rulegroup @defaults("name") {
+  // ARN of the rulegroup
+  arn string
+  // ID of the rulegroup
+  id string
+  // Name of the rulegroup
+  name string
+  // Description of the rulegroup
+  description string
+  // List of waf rules
+  rules() []aws.waf.rule
+}
+
+// Amazon WAF rule
+private aws.waf.rule @defaults("name") {
+  // arn of the acl/ruleGroup + the name of the rule
+  id string
+  // Name of the rule
+  name string
+  // Priority from lowest to highest number
+  priority int
+  // Part of the rule that tells WAF how to inspect a web request
+  statement aws.waf.rule.statement
+  // Part of the rule that tells WAF what to do with a web request when it matches the criteria defined in the rule
+  action aws.waf.rule.action
+  // The ARN of either rule ACL or the RuleGroup that this rule belongs to
+  belongsTo string
+}
+
+// Action that happens if a rule statement matches
+private aws.waf.rule.action @defaults("action") {
+  // Name of the rule this action belongs to
+  ruleName string
+  // One of Block, Allow, Count, Captcha
+  action string
+  // HTTP Response Code, only if the action is Block
+  responseCode string
+}
+
+private aws.waf.rule.statement @defaults("kind") {
+  // random uuid, to make the statement unique, these don't exist in AWS
+  id string
+  kind string
+  sqliMatchStatement aws.waf.rule.statement.sqlimatchstatement
+  xssMatchStatement aws.waf.rule.statement.xssmatchstatement
+  byteMatchStatement aws.waf.rule.statement.bytematchstatement
+  regexMatchStatement aws.waf.rule.statement.regexmatchstatement
+  geoMatchStatement aws.waf.rule.statement.geomatchstatement
+  ipSetReferenceStatement aws.waf.rule.statement.ipsetreferencestatement
+  labelMatchStatement aws.waf.rule.statement.labelmatchstatement
+  managedRuleGroupStatement aws.waf.rule.statement.managedrulegroupstatement
+  notStatement aws.waf.rule.statement.notstatement
+  orStatement aws.waf.rule.statement.orstatement
+  andStatement aws.waf.rule.statement.andstatement
+  rateBasedStatement aws.waf.rule.statement.ratebasedstatement
```

Comment:
this really is a perfect example of where we could use union types. 
but lacking union types, what about adding a "resolvedStatementJson" field that just puts the statement into a json dict, so folks can see the thing easily without first knowing the kind?

Reply:
```
aws.waf.acls.first.rules.first.statement.json: {
  AndStatement: null
  ByteMatchStatement: null
  GeoMatchStatement: null
  IPSetReferenceStatement: null
  LabelMatchStatement: null
  ManagedRuleGroupStatement: null
  NotStatement: null
  OrStatement: null
  RateBasedStatement: null
  RegexMatchStatement: null
  RegexPatternSetReferenceStatement: null
  RuleGroupReferenceStatement: null
  SizeConstraintStatement: null
  SqliMatchStatement: {
    FieldToMatch: {
      AllQueryArguments: null
      Body: {
        OversizeHandling: "MATCH"
      }
      Cookies: null
      HeaderOrder: null
      Headers: null
      JA3Fingerprint: null
      JsonBody: null
      Method: null
      QueryString: null
      SingleHeader: null
      SingleQueryArgument: null
      UriPath: null
    }
    SensitivityLevel: "LOW"
    TextTransformations: [
      0: {
        Priority: 0.000000
        Type: "NONE"
      }
    ]
  }
  XssMatchStatement: null
}
```